### ISSUE TITLE
welcome page(startify)  icon can't display

### ISSUE THREAD

**Issue by zgpio** [wsdjeg/SpaceVim#1701]
## Expected behavior
startify filetype icon display correctly
## The reproduce ways from Vim starting (Required!)
On Win10, I clone spacevim to my  home dir according to wiki.
I install [nerd font](https://github.com/ryanoasis/nerd-fonts) from it [release page](https://github.com/ryanoasis/nerd-fonts/releases).
I use `set guifont=*` to choose font.
After that gvim can display statusline correctly but not the startify filetype icon.
## Debug info

I use defalut spacevim's config

## Screenshots
It shows like `<a8><9f>`, just like the code of font.
![default](https://user-images.githubusercontent.com/19503791/39818719-24236c12-53d4-11e8-8654-b9993f9ddaac.png)
And the tabline icon:
![default](https://user-images.githubusercontent.com/19503791/39964433-1ae348b2-56b7-11e8-9bc4-6b79230d4043.png)

**Comment by wsdjeg** [comment]
Which filetype?

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39846562-ea92db44-542e-11e8-9500-4e439a844e43.png)
and the fileformat icon is "?",
![default](https://user-images.githubusercontent.com/19503791/39846880-b75c6932-5430-11e8-8c81-2e48547d5c3d.png)

**Comment by wsdjeg** [comment]
Can you check the icon on tabline?

**Comment by wsdjeg** [comment]
![2018-05-11-20 18 08](https://user-images.githubusercontent.com/13142418/39924230-9a58fa1e-54ec-11e8-9bd1-09b831679669.png)

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39924401-3113359c-555a-11e8-8b30-38dd3641a5cc.png)
no tabline icon

**Comment by wsdjeg** [comment]
you need to enable filetype icon in tabline.

**Comment by zgpio** [comment]
It default enable. 
![default](https://user-images.githubusercontent.com/19503791/39951364-c7b4b456-55bb-11e8-8397-2ac520d68271.png)

**Comment by wsdjeg** [comment]
It should be:

`    enable_tabline_filetype_icon = true`

**Comment by zgpio** [comment]
terrible
![default](https://user-images.githubusercontent.com/19503791/39963286-f5e44bd2-5699-11e8-9336-a682a032803d.png)
I use nerd font， my own vim dotfile is work， but SpaceVim ...

**Comment by wsdjeg** [comment]
@zgpio I just reopen this issue, It seem issue with your fonts

This is issue with your fonts, your fonts does not included this icon, cam you run 

```vim
:echo SpaceVim#api#file#get().fticon('foo.vim')
```

I think you should see 

![2018-05-13_115x55](https://user-images.githubusercontent.com/13142418/39964644-b13ead42-564e-11e8-81d3-cbc92c2b391a.png)

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39964672-38824ac6-56bc-11e8-807e-815bbb28f8ce.png)
But I see < a8 > < 9f > code.
I only install nerd font from the [nerd font release page](https://github.com/ryanoasis/nerd-fonts)

**Comment by wsdjeg** [comment]
sorry, I can not fix this issue, nothing can be done with SpaceVim, This is issue with fonts.

**Comment by zgpio** [comment]
I don‘t think it is a issue with fonts

![default](https://user-images.githubusercontent.com/19503791/39965403-a3032f06-56ca-11e8-87ec-05f8b8a63e6c.png)

With this font set，I can see the vim icon in the buffer.
Why I can't see it in the startify or tabline？

**Comment by wsdjeg** [comment]
can you select that char, and copy it to clipboard, then use `echo @"` to see if it can be displayed?

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39965484-363de030-56cc-11e8-978c-1cec30d42202.png)
it is right

**Comment by wsdjeg** [comment]
how about `echo SpaceVim#api#file#get().fticon('foo.vim') == @"`?

**Comment by zgpio** [comment]
false（0）

**Comment by wsdjeg** [comment]
Ok, It seem the api is brokend. I will add some log for this API tonight, hope you can help me to test it. It is 16:40 PM, the new branch will be pushed before 22:00.

**Comment by zgpio** [comment]
I will do my best to help you.

**Comment by wsdjeg** [comment]
what is the result of `echo SpaceVim#api#file#get().fticon('foo.vim') == "\xa8\x9f"`

**Comment by zgpio** [comment]
true(1)

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39966113-07632254-56d9-11e8-9a6e-dab66b9c178b.png)

![default](https://user-images.githubusercontent.com/19503791/39966117-220aaaaa-56d9-11e8-832b-093f6e3a2502.png)

str1 work, but not str2

**Comment by zhaozg** [comment]
are your file utf-8 encoded？

**Comment by zgpio** [comment]
yes, this is the /autoload/SpaceVim/api/file.vim

**Comment by wsdjeg** [comment]
@zgpio with the test from your script, I think no need to add log, the issue is your vim can not display `str2`, but `str1` works, maybe we need to use `"\ue7c5"` instead of use icon.

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39966709-2797e80c-56e3-11e8-8ae6-b9c903494849.png)
but it can't work, I can't solve it.

it display like this:
![default](https://user-images.githubusercontent.com/19503791/39966723-4f1cf3fe-56e3-11e8-9012-93bc7725cadb.png)

And I have a problem about why linux can work.

**Comment by wsdjeg** [comment]
can you just open the file.vim, and move your cursor onto the vim filetype icon, then press `ga`.

you should see:

![2018-05-13_318x44](https://user-images.githubusercontent.com/13142418/39966737-78ba7636-5676-11e8-8616-91c2f70fd8af.png)

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39966752-9c617928-56e3-11e8-854e-c7b90a55c4a1.png)
yes I saw it

**Comment by wsdjeg** [comment]
Sorry, I can not fix it, maybe choose a different font would works well, and what is your vim version, gvim or vim? and what is font do you use? `echo &guifont` ?

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/39966812-53ebba5e-56e4-11e8-9419-bbe077fd7eca.png)
`vim8.0` `gvim`

**Comment by wsdjeg** [comment]
@zgpio I can not fix this issue, do you have a try with vim-devicons does that plugin works well in your computer？

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/41017983-30bcceb6-698a-11e8-9f65-507df1b58001.png)
It works on my vim config.
Other windows user meet this problem?

**Comment by wsdjeg** [comment]
@zgpio you do not understand my question,  in spacevim the vim filetype icon can not displayed, so you need to test with vim file in vim-devicons instead of zgp and config file.

**Comment by zgpio** [comment]
I put the pack/git/opt/vim-devicons, and then change as follow, It can display icons:
我在spacevim根目录放置了pack/git/opt/vim-devicons, 然后改变如下，这样可以工作：
![default](https://user-images.githubusercontent.com/19503791/41020114-3e055c14-6994-11e8-89f7-618c4c9682a7.png)

as you can see：
![default](https://user-images.githubusercontent.com/19503791/41020248-ac4ca4d4-6994-11e8-98cf-c882c0fc407f.png)
请问只有我提过这样的问题吗？

**Comment by wsdjeg** [comment]
this icon is for `*.md` file so how about `*.vim` file?

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/41020622-2fe1c8a0-6996-11e8-90b6-8b87712a0108.png)

**Comment by zgpio** [comment]
Maybe the implement of vim-devicons can answear this problem！But I don't have much time.

**Comment by wsdjeg** [comment]
@zgpio  Ok, I will check it, thanks for your info.

**Comment by zgpio** [comment]
thanks a lot!

**Comment by ssfdust** [comment]
换到neovim试试。

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/41140202-91788f48-6b1e-11e8-854f-86406c9f98a4.png)
Actually，I tested it, neovim work，but I use gvim, not neovim.

**Comment by wsdjeg** [comment]
@zgpio do you test all the filetype? only the vim filetype icon does not work?

**Comment by wsdjeg** [comment]
@zgpio the plugin vim-devicons use different icon group, This is issue with font, I can not fix it, but you can change the icon via spacevim option `g:spacevim_filetype_icons`.

for example, if you want to change the vim file icon:

```vim
let g:spacevim_filetype_icons['vim'] = ''
```

if you want to change the markdown file type icon:

```vim
let g:spacevim_filetype_icons['md'] = ''
```

**Comment by zgpio** [comment]
I put the pack/git/opt/vim-devicons, and then change as follow, It can display icons:
```bash
diff --git a/config/plugins/vim-startify.vim b/config/plugins/vim-startify.vim
index b5d23ae..d61479c 100644
--- a/config/plugins/vim-startify.vim
+++ b/config/plugins/vim-startify.vim
@@ -69,8 +69,10 @@ if g:spacevim_enable_tabline_filetype_icon
     return empty(icon) ? ' ' : icon
   endfunction

+  packadd vim-devicons
   function! StartifyEntryFormat()
-    return 'FileIcon(entry_path) ."  ". entry_path'
+    "return 'FileIcon(entry_path) ."  ". entry_path'
+    return 'WebDevIconsGetFileTypeSymbol(absolute_path) ." ". entry_path'
   endfunction

 endif
```
With the font I use, the icons in `file.vim` display well, but the  icon  of  tabline/statuline/startify can't work.
As follow:
![default](https://user-images.githubusercontent.com/19503791/42862514-66ae7ab4-8a92-11e8-94e1-79ec07388fe3.png)
![default](https://user-images.githubusercontent.com/19503791/42862535-7ac3e55c-8a92-11e8-855b-0b1058c45253.png)


Can't be fixed on 0.9.0 dev?

**Comment by motecshine** [comment]
may be your terminal font was not patched
